perfkitbenchmarker/pkb.py

```diff
@@ -428,13 +436,46 @@ def RunBenchmarks(publish=True):
       vm_util.RunThreaded(
           RunBenchmark, args, max_concurrent_threads=FLAGS.parallelism)
     else:
-      for run_args, kwargs in args:
-        RunBenchmark(*run_args, **kwargs)
+      while args:
+        run_args, _ = args.popleft()
+        benchmark_module, _, sequence_number, _, _, benchmark_uid = run_args
+        benchmark_name = benchmark_module.BENCHMARK_NAME
+        try:
+          RunBenchmark(*run_args)
+          benchmark_arg_succeeded_tuples.append((run_args, True))
+        except BaseException as e:
+          benchmark_arg_succeeded_tuples.append((run_args, False))
+          msg = 'Benchmark {0}/{1} {2} (UID: {3}) failed.'.format(
+              sequence_number, total_benchmarks, benchmark_name, benchmark_uid)
+          if (isinstance(e, KeyboardInterrupt) or
+              FLAGS.stop_after_benchmark_failure):
+            logging.error('%s Execution will not continue.', msg)
+            break
+          logging.error('%s Execution will continue.', msg)
+      benchmark_arg_succeeded_tuples.extend((a, False) for a, _ in args)
 
   finally:
     if collector.samples:
       collector.PublishSamples()
 
+    if benchmark_arg_succeeded_tuples:
```

Comment:
No, this is different. benchmark_arg_succeeded_tuples is a list of (args tuple, succeeded boolean) tuples, one per benchmark. It includes results for successful and failed benchmarks in order of execution per the other comment suggestion. But it is not populated in some cases: if --parallelism is used (since that's pending deletion and may not be worth the trouble to implement) or if, for some reason, the user specifies --benchmarks="".